### PR TITLE
Feature/common io t

### PR DESCRIPTION
**Pull request by ChristopherRogers1991**
## Description
This add a CommonIotSkill base class, that will be used by skill that work with the CommonIot framework.

This and the two skills linked below are still works in progress. Combined they implement the core, initial CommonIot framework.

## How to test
Load https://github.com/MycroftAI/iot-test-skill and https://github.com/MycroftAI/skill-iot-control and issue commands like "turn on the lights," "toggle the kitchen lights," and "bedroom off."

## Contributor license agreement signed?
CLA [ ] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: mycroft/skills/common_iot_skill.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
+# Copyright 2018 Mycroft AI Inc.
```
Comment (forslund) [inline mycroft/skills/common_iot_skill.py]: 2019?

File: mycroft/skills/common_iot_skill.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+# BE WARNED THAT THE CLASSES, FUNCTIONS, ETC MAY CHANGE WITHOUT WARNING.
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from mycroft import MycroftSkill
+from mycroft.messagebus.message import Message
+from mycroft.util.log import getLogger
+
+
+LOGGER = getLogger(__name__)
```
Comment (forslund) [inline mycroft/skills/common_iot_skill.py]: We usually do 

```python
from mycroft.util.log import LOG
```
and then just use

```python
LOG.info('This is useful information')
```

**Comment by ChristopherRogers1991** [comment]
Sorry, I assumed everyone in the org would have access to those repos by default. They are very rough and not ready for public consumption yet, but I did intend for the core team to have access. I've added you (and the Mycroft core team) to the repos, so you should have access now.

I'll address the other nits a little later this evening.

**Comment by forslund** [comment]
Thanks now I can access. Seems to work fine as far as I can see.

**Comment by forslund** [comment]
If @penrods doesn't have any other objections I'll merge it tomorrow morning after you've got a chance to address those tiny things.

File: mycroft/skills/common_iot_skill.py (hunk 1 above)
Comment (ChristopherRogers1991) [reply mycroft/skills/common_iot_skill.py]: Fixed.

File: mycroft/skills/common_iot_skill.py (hunk 2 above)
Comment (ChristopherRogers1991) [reply mycroft/skills/common_iot_skill.py]: Removed this. The logger was only being used for some debugging yesterday, and I just forgot to remove it.

**Pull request merged**